src/bitmaps.rs

```diff
@@ -1,202 +1,97 @@
-use std::borrow::Borrow;
+use std::borrow::Cow;
 use std::fs;
 use std::fs::File;
 use std::io::BufWriter;
 use std::path::Path;
 
-use bitreader::{BitReader, BitReaderError};
-
 use allsorts::binary::read::ReadScope;
-use allsorts::bitmap::{self, BitDepth, CBDTTable, CBLCTable, GlyphBitmapData, MatchingStrike};
-use allsorts::fontfile::FontFile;
-use allsorts::tables::FontTableProvider;
-use allsorts::tag;
+use allsorts::bitmap::{BitDepth, Bitmap, BitmapGlyph, EncapsulatedFormat};
+use allsorts::font_data::FontData;
+
+use allsorts::Font;
 
 use crate::cli::BitmapOpts;
 use crate::BoxError;
+use allsorts::font::MatchingPresentation;
+use allsorts::tag::DisplayTag;
 
 pub fn main(opts: BitmapOpts) -> Result<i32, BoxError> {
     let buffer = std::fs::read(&opts.font)?;
     let scope = ReadScope::new(&buffer);
-    let font_file = scope.read::<FontFile>()?;
+    let font_file = scope.read::<FontData>()?;
     let table_provider = font_file.table_provider(opts.index)?;
-
-    let table = table_provider
-        .table_data(tag::CBLC)?
-        .or(table_provider.table_data(tag::EBLC)?)
-        .ok_or("font does not have CBLC or EBLC tables")?;
-    let scope = ReadScope::new(table.borrow());
-    let cblc = scope.read::<CBLCTable<'_>>().unwrap();
-    let table = table_provider
-        .table_data(tag::CBDT)?
-        .or(table_provider.table_data(tag::EBDT)?)
-        .ok_or("font does not have CBDT or EBDT tables")?;
-    let scope = ReadScope::new(table.borrow());
-    let cbdt = scope.read::<CBDTTable<'_>>().unwrap();
+    let mut font = Font::new(table_provider)?.ok_or("unable to find suitable cmap sub-table")?;
 
     let output_path = Path::new(&opts.output);
     if !output_path.exists() {
         fs::create_dir(output_path)?;
     }
 
-    for strike in &cblc.bitmap_sizes {
-        let strike = &strike.inner;
-        let strike_path = output_path.join(&format!(
-            "{}x{}@{}",
-            strike.ppem_x, strike.ppem_y, strike.bit_depth as u8
-        ));
-        if !strike_path.exists() {
-            fs::create_dir(&strike_path)?;
+    for ch in opts.text.chars() {
+        let (glyph_id, _) = font.lookup_glyph_index(ch, MatchingPresentation::NotRequired, None);
+        if glyph_id == 0 {
+            eprintln!("No glyph for '{}'", ch);
+            continue;
         }
-        for glyph_id in strike.start_glyph_index..=strike.end_glyph_index {
-            if let Some(matching_strike) =
-                cblc.find_strike(glyph_id, strike.ppem_x, strike.bit_depth)
-            {
-                dump_bitmap(matching_strike, glyph_id, &strike_path, &cbdt)?;
-            }
-        }
-    }
 
-    Ok(0)
-}
+        match font.lookup_glyph_image(glyph_id, opts.size, BitDepth::ThirtyTwo)? {
+            Some(bitmap) => {
+                let strike_path = output_path.join(&format!(
+                    "{}x{}",
```

Comment:
Sorry, I'm not familiar with the details, but just confirming we're going from `"{}x{}@{}"` to `"{}x{}"`?

Reply:
My bad. I should have noted that I reworked the bitmap extraction sub-command. It much simpler because it goes through a higher level API but that also means it's not quite as flexible. The end result is that the `bitmap` command will now extract a single strike size at a time so there's not need to differentiate with `@{}`.